File: src/main/java/ru/citeck/ecos/predicate/model/AttributePredicate.java
package ru.citeck.ecos.predicate.model;

import ru.citeck.ecos.predicate.type.Predicate;

public class AttributePredicate implements Predicate {

    private String attribute;

    public String getAttribute() {
        return attribute;
    }

    public void setAttribute(String attribute) {
        this.attribute = attribute;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records2/predicate/model/AttributePredicate.kt
package ru.citeck.ecos.records2.predicate.model

import com.fasterxml.jackson.annotation.JsonProperty

abstract class AttributePredicate : Predicate {

    @JsonProperty("att")
    private var attribute: String = ""

    @JsonProperty("a")
    fun setAtt(attribute: String?) {
        setAttribute(attribute)
    }

    fun setAttribute(attribute: String?) {
        this.attribute = attribute ?: ""
    }

    fun getAttribute(): String {
        return attribute
    }
}
